surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/RunListenerAdapter.java

```diff
@@ -259,4 +261,50 @@ else if ( testSource.filter( ClassSource.class::isInstance ).isPresent() )
             return new String[] { source, source, display, display };
         }
     }
+
+    public String[] toClassMethodNameWithoutPlan( TestIdentifier testIdentifier )
```

Comment:
I am not totally convinced about this method because there is already one which returns `String[4]`. I would vote for a generic approach avoiding code duplication in single class.

Reply:
Are you referring to `toClassMethodName`? That couldn't be used because at the time of requesting class names for reruns the plan is set to `null`. Perhaps a null check in the original method would suffice or were you thinking of a different approach?